mealie/services/scheduler/scheduler_service.py

```diff
@@ -42,14 +42,20 @@ async def schedule_daily():
         hour_target = 23
         minute_target = 45
 
-    hours_until = ((hour_target - now.hour) % 24) or 24
-    minutes_until = (minute_target - now.minute) % 60
+    next_schedule = now.replace(hour=hour_target, minute=minute_target, second=0, microsecond=0)
+    delta = next_schedule - now 
+    if delta < timedelta(0):
+        next_schedule = next_schedule + timedelta(days=1)
+        delta = next_schedule - now
+
+    hours_until = int(delta.total_seconds() / 3600)
+    minutes_until = int(delta.total_seconds() % 3600 / 60)
     logger.debug("Hours until %s and minutes until %s", str(hours_until), str(minutes_until))
```

Comment:
Firstly, thanks for raising the PR @vladyslavq, much appreciated!

I haven't (yet) had time to look too closely, but your implementation looks good!

One oddity I've noticed however is this line displays oddly, which I think is a result of the fact that you don't actually care about the `minutes_until` variable, it's only used for display in the modified version. Take this example in my dev container:
![image](https://github.com/mealie-recipes/mealie/assets/3479092/b7a210f2-e4d4-4f70-961d-16471a887f7e)

The resulting calculation is correct, but intuitively it would be 19 minutes, not 18.

Maybe given your change to the overall logic, this `logger.debug` output would be best outputting things related to the `delta` variable, and not trying to conform to how the old message was structured?

Reply:
@boc-the-git  Sure, just didn't want to change anything outside the logic. Please check the new commit.